tracer/test/test-applications/integrations/Samples.InstrumentedTests/Vulnerabilities/String/StringFormatTests.cs

```diff
@@ -0,0 +1,353 @@
+using System;
+using System.Text;
+using Xunit;
+
+namespace Samples.InstrumentedTests.Iast.Vulnerabilities.StringPropagation;
+public class StringFormatTests : InstrumentationTestsBase
+{
+    private string _taintedValue = "tainted";
+    private string _taintedValue2 = "TAINTED2";
+    private string _taintedFormat2Args = "format{0}{1}";
+    private string _taintedFormat3Args = "format{0}{1}{2}";
+    private string _taintedFormat1Arg = "format{0}";
+    private string _untaintedString = "UntaintedString";
+    private string _otherUntaintedString = "OtherUntaintedString";
+
+    public StringFormatTests()
+    {
+        AddTainted(_taintedValue);
+        AddTainted(_taintedFormat1Arg);
+        AddTainted(_taintedFormat2Args);
+        AddTainted(_taintedFormat3Args);
+        AddTainted(_taintedValue2);
+    }
+
+    // Testing public static string Format(IFormatProvider provider, string format, params object[] args)
+
+    [Fact]
+    public void GivenATaintedObject_WhenCallingFormatWithProvider_ResultIsTainted()
+    {
+        AssertTaintedFormatWithOriginalCallCheck(":+-test: tainted-+:", 
+            String.Format(new FormatProviderForTest(), "test: {0}", new object[] { _taintedValue }), 
+            () => String.Format(new FormatProviderForTest(), "test: {0}", new object[] { _taintedValue }));
+    }
+
+    [Fact]
+    public void GivenATaintedObject_WhenCallingFormatWithProvider_ResultIsTainted2()
```

Comment:
these tests could maybe be grouped under `Theories` receiving `MemberData` for readbility